R/pmcmc.R

```diff
@@ -0,0 +1,1029 @@
+      propose_jump = propose_jump,
+      first_data_date = data$date[1],
+      output_proposals = output_proposals,
+      .progress = TRUE)
+  }
+
+  #..................
+  # MCMC diagnostics and tidy
+  #..................
+  if (n_chains > 1) {
+    names(chains) <- paste0('chain', seq_len(n_chains))
+
+    # calculating rhat
+    # convert parallel chains to a coda-friendly format
+    chains_coda <- lapply(chains, function(x) {
+
+      traces <- x$results
+      if('start_date' %in% names(pars_init)) {
+        traces$start_date <- start_date_to_offset(data$date[1], traces$start_date)
+      }
+
+      coda::as.mcmc(traces[, names(pars_init)])
+    })
+
+    rhat <- tryCatch(expr = {
+      x <- coda::gelman.diag(chains_coda)
+      x
+    }, error = function(e) {
+      print('unable to calculate rhat')
+    })
+
+
+    pmcmc <- list(inputs = chains[[1]]$inputs,
+                  rhat = rhat,
+                  chains = lapply(chains, '[', -1))
+
+    class(pmcmc) <- 'squire_pmcmc_list'
+  } else {
+    pmcmc <- chains[[1]]
+    class(pmcmc) <- "squire_pmcmc"
+  }
+  #............................................................
+  # Section 3 of pMCMC Wrapper: Sample PMCMC Results
+  #...........................................................
+  pmcmc_samples <- sample_pmcmc(pmcmc_results = pmcmc,
+                                burnin = burnin,
+                                n_chains = n_chains,
+                                n_trajectories = replicates,
+                                n_particles = n_particles,
+                                forecast_days = forecast)
+
+  #............................................................
+  # Section 4 of pMCMC Wrapper: Tidy Output
+  #...........................................................
+  #..................
+  # Pull Sampled results and "recreate" squire models
+  #..................
+  if (inherits(squire_model, "stochastic")) {
+
+    # create a fake run object and fill in the required elements
+    r <- squire_model$run_func(country = country,
+                               contact_matrix_set = contact_matrix_set,
+                               tt_contact_matrix = tt_contact_matrix,
+                               hosp_bed_capacity = hosp_bed_capacity,
+                               tt_hosp_beds = tt_hosp_beds,
+                               ICU_bed_capacity = ICU_bed_capacity,
+                               tt_ICU_beds = tt_ICU_beds,
+                               population = population,
+                               replicates = 1,
+                               time_period = max(tt_contact_matrix,tt_hosp_beds,tt_ICU_beds,1))
+
+    # first let's add our pmcmc results for a nice return
+    # we'll save our inputs so it is easy to recreate later
+    r$pmcmc_sample_inputs <- pmcmc_samples$inputs
+    # and add the parameters that changed between each simulation, i.e. posterior draws
+    r$trajectory_parameters <- pmcmc_samples$sampled_PMCMC_Results
```

Comment:
change this to `replicate_parameters` and it should play nicer with `projections`

Reply:
cheers!